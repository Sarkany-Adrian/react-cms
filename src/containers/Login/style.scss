.login-wrapper {
  width: 320px;
  margin: 0 auto;
  margin-top: 30px;
  text-align: center;
  padding: 20px;
}
