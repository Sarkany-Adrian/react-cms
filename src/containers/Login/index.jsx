// @flow
import React, { PureComponent } from 'react';
// style
import './style.scss';
// atoms
import Btn from 'atoms/Btn';

class Login extends PureComponent<{}> {
  handleClick = (e: SyntheticKeyboardEvent<*>) => {
    e.preventDefault(); // just a placeholder
  };

  render() {
    return (
      <div className="login-wrapper">
        <p>login page</p>
        <Btn handleClick={this.handleClick}>Login</Btn>
      </div>
    );
  }
}

export default Login;
