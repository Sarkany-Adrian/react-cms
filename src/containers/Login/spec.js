import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Login from 'containers/Login';

const auth = {
  isLoggedIn: false,
  setLoggedIn: jest.fn()
};

describe('Login container', () => {
  it('should render the login container', () => {
    const wrapper = shallow(<Login.wrappedComponent auth={auth} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
