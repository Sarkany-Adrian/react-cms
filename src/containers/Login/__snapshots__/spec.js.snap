// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login container should render the login container 1`] = `
<div
  className="login-wrapper"
>
  <p>
    Login to continue:
  </p>
  <Button
    className=""
    onClick={[Function]}
    rest={null}
  >
    Login
  </Button>
</div>
`;
